heudiconv/utils.py

```diff
@@ -222,6 +223,41 @@ def load_json(filename: str | Path, retry: int = 0) -> Any:
     return data
 
 
+def is_deprecated_seriesid(series_id: str, len_hash_hex: int = 8) -> bool:
+    """Return True if series_id is in deprecated format.
+    A deprecated series ID is in the format "series_number-protocol_name".
+    A non-deprecated series ID is in the format "series_number-protocol_name-seriesUID_hash_hex"
+    """
+    # Check at least two '-' in the series_id
+    if series_id.count('-') <= 1:
+        return True
+    # Check the first part of the series_id is a number
+    series_number = series_id.split('-')[0]
+    if not series_number.isdigit():
+        return True
+    # Check the last part of the series_id is a hash in hexadecimal format of length len_hash_hex
+    hash_hex = series_id.split('-')[-1]
+    hex_digits = set(string.hexdigits)
+    if len(hash_hex) != len_hash_hex or not all(c in hex_digits for c in hash_hex):
+        return True
+    # If all previous tests passed, then the series_id is not deprecated
+    return False
```

Comment:
```suggestion
    return len(hash_hex) != len_hash_hex or not all(c in hex_digits for c in hash_hex)
```

Reply:
or may be better to use `re` (might need an import)

```suggestion
    return re.match(f"[{string.hexdigits}]{{{len_hash_hex}}}$")
```

this function needs a unittest which would be trivial to do